### DISCUSSION TITLE
Support for media player assumed state

### DISCUSSION THREAD

**Discussion by thecode** [home-assistant/architecture#720] [Entity Models] [answered]
# Background
Currently media player supports the following states to describe a device that is active (not off or unknown):
`STATE_ON`, `STATE_PAUSED`, `STATE_IDLE`, `STATE_PLAYING`
Some reference for how the frontend display different states of the entity:
- `STATE_ON`: 
![image](https://user-images.githubusercontent.com/1858925/151451865-7cea7bbf-3568-4a2b-8a79-61936f48bb98.png)
- `STATE_IDLE`
![image](https://user-images.githubusercontent.com/1858925/151452309-21b0b708-f9ec-43a3-b2cf-a7294fbb9bc7.png)
- `STATE_PAUSED`:
![image](https://user-images.githubusercontent.com/1858925/151451981-7e132c88-4b54-491d-bedc-32a9e86a6158.png)
- `STATE_PLAYING`
![image](https://user-images.githubusercontent.com/1858925/151452541-51192578-8458-4ecf-9fb0-ee59145d18e4.png)

To simplify the discussion I left out the off/unknown states and the power button frontend behavior, it is also assumed that the integration implementing the media player knows if the device is on or off.

# Current behavior
Many media players devices (I counted 25, but there might be more which implement a virtual playing/paused state) does not know if the device is currently playing something, if it is in stop/play/paused state and declare `STATE_ON`, some try to implement a virtual state by assuming if the device is playing or paused and reversing the state when frontend call play or pause, this however may loss sync with the actual state of the device.

Prior to https://github.com/home-assistant/frontend/pull/8845 integrations reporting `STATE_ON` would show the following:
![image](https://user-images.githubusercontent.com/1858925/151451865-7cea7bbf-3568-4a2b-8a79-61936f48bb98.png)
The fronted would call `media_play_paused`, this would somehow allow setting the play/pause state of the device but miss the next/previous buttons. The `media_play_paused` service was not implemented by all integrations and would produce errors since there is also no `supported_features` flag to indicate if it is supported. https://github.com/home-assistant/frontend/pull/8845 changed the behavior to call `media_play` service which must send play command to the device and can't implement virtual play/paused as implemented by many integrations.

# Suggestion
Add a `supported_features` flag `SUPPORT_ONLY_DUMB_STATE` which when indicated by the integration will disable frontend logic and will show the following for `STATE_ON`:
![image](https://user-images.githubusercontent.com/1858925/151454504-b95247ac-30a6-46ba-8ffd-b87579250d9d.png)
The `supported_features` will still indicate which services are available so that if for example `SUPPORT_PAUSE` is not set by the integration the button would not show. This will allow operating devices that does not have feedback of the current playing state (hence **"DUMB"**) with better user experience. `STATE_ON` would still be applicable here since it is the only state we know of the device.

EDIT: Instead of adding a `SUPPORT_ONLY_DUMB_STATE` flag we can use the `assumed_state` property, but the logic above is still required.

### Relevant links:
- https://github.com/home-assistant/core/issues/64172
- https://github.com/home-assistant/core/issues/44583
- [Media Player Entity](https://developers.home-assistant.io/docs/core/entity/media-player/)
- https://github.com/home-assistant/architecture/discussions/515

**Comment by MartinHjelmare** [comment, marked as answer]
The existing way for an entity to indicate that it doesn't support device feedback on actions is the property `assumed_state`.

**Reply by thecode** [reply]
I edited the description to indicate we can use the `assumed_state` instead of a new flag, but the logic above is still required

**Reply by MartinHjelmare** [reply]
It's a pure frontend adjustment, right?

**Reply by thecode** [reply]
I am still not sure how we still handle the `STATE_ON` and `STATE_OFF` in this case, but if the frontend it changed so that in `assumed_state` it will allow all controls it is possible to solve it only by changing the frontend behavior

**Reply by balloob** [reply]
I like using assumed state to implement this.

**Reply by balloob** [reply]
Here is a link to the frontend code https://github.com/home-assistant/frontend/blob/dev/src/data/media-player.ts#L231

**Reply by thecode** [reply]
Implemented at: https://github.com/home-assistant/frontend/pull/11642

**Reply by elupus** [reply]
I know it's closed. But we have discovered a problem with this "change". Some media players do know their on/off state, but they don't know their playback state (playing/paused). I'm just posting it here for memory. Will need a new arch discussion to handle.
